src/ng/directive/form.js

```diff
@@ -490,7 +513,9 @@ var formDirectiveFactory = function(isNgForm) {
               });
             }
 
-            var parentFormCtrl = controller.$$parentForm;
+            var parentFormCtrl = ctrls[1] || ctrls[0].$$parentForm;
+            parentFormCtrl.$addControl(controller);
```

Comment:
Hard-coding `parentFormCtrl`suffers from the same issues in [`attr.$observe(...)`][1] and [`formElement.on('$destroy', ...)`][2]; i.e. if the form has been since removed and re-added to another parent form, it won't be properly renamed/removed.

You need to use `ctrls[0].$$parentForm`.

(It's OK to use `parentFormCtrl` on line 517 though.)

[1]: https://github.com/angular/angular.js/pull/12418/files#diff-1838ed9bd28307b6d42de98549826ae5R523
[2]: https://github.com/angular/angular.js/pull/12418/files#diff-1838ed9bd28307b6d42de98549826ae5R531